Fix issues with deprecated APIs and changed frame elements (fixes for 8.2.5)

mudohir opened the pull request:
Made new function to call BNGetFriendInfo or C_BattleNet.GetFriendAccountInfo() depending on retail/classic
Merged group title into nameText for dividers (text element no longer present in divider) for retail where the text has been removed from friends list buttons.
Made an addon-wide variable to generalize between FriendsFrameFriendsScrollFrame and FriendsListFrameScrollFrame - different frame names are used in classic/retail
Call OnEnter() for buttons instead of removed FriendsFrameTooltip_Show()
Fix some whitespace unifgit statgiormity issues (spaces to tabs)
Add new HookButtons() that hooks the "OnClick" function for each button with FriendGroups_OnClick
Fixed issue with right click on dividers.
Made sure code worked on retail and classic

Mikeprod commented:
Hello, Thanks a lot, I also did some local changes localy, but could not figure out some things not properly working. I will review your PR asap. Hopefully this weekend because the community is eager to get an update on this.

DJDajn commented:
Yo, is it possible to just change line 314 from
`- button.name:SetJustifyH("RIGHT")` to
`+ button.name:SetJustifyH("LEFT")`?
Looks alot better UI wise in my opinion

nuresproc commented:
Dear All,
Thanks for the fixing.
Glad to have a prototype can use at this moment. It wont shows correctly for friends locate/instance it shows "Unknow" but glad that i finally can fold it.

mudohir commented:
> Yo, is it possible to just change line 314 from
> `- button.name:SetJustifyH("RIGHT")` to
> `+ button.name:SetJustifyH("LEFT")`?
> Looks alot better UI wise in my opinion
> 
> Also, It doesn't return the current zone/area in the UI, so right now it will either show nothing when in another game, app/mobile or it will just show unknow when playing WoW (Picture below).
> https://imgur.com/eDirjYK

Made a new commit justifying the dividers to CENTER to match how it works in classic - author may decide to change that to be LEFT instead.

Think I've fixed issues with zone names/realm names.

Also updated colors for classic so if you're there you should see class colors properly.

mudohir commented:
> Dear All,
> Thanks for the fixing.
> Glad to have a prototype can use at this moment. It wont shows correctly for friends locate/instance it shows "Unknow" but glad that i finally can fold it.

Think newest commit fixes this issue let me know if it is still broken for you.

mudohir commented:
> I'm happy to hear it's been fixed, but I don't speak code at all, what am I to do with all this ?
> What do I click to get this new file ?
> Do I have to wait for this to come on twitch ?

You should be able to go to the files for the pull request:

https://github.com/Mikeprod/FriendGroups/pull/41/files

From here click on the "..." near the top of the file on the right and then select "view file".
You can copy the contents here into the FriendGroups.lua inside the addon directory for your wow installation.

There may be a way way easier way to do this in the UI, but I am not super familiar with github UI.

DJDajn commented:
> > I'm happy to hear it's been fixed, but I don't speak code at all, what am I to do with all this ?
> > What do I click to get this new file ?
> > Do I have to wait for this to come on twitch ?
> 
> You should be able to go to the files for the pull request:
> 
> https://github.com/Mikeprod/FriendGroups/pull/41/files
> 
> From here click on the "..." near the top of the file on the right and then select "view file".
> You can copy the contents here into the FriendGroups.lua inside the addon directory for your wow installation.
> 
> There may be a way way easier way to do this in the UI, but I am not super familiar with github UI.

Yea, everything looks right now with the zones/app info :)

mudohir commented:
> Hello, Thanks a lot, I also did some local changes, but could not figure out some things not properly working. I will review your PR asap. Hopefully this weekend because the community is eager to get an update on this.

Hey, I am not 100% happy with how I have implemented, but it's probably fine for now.
After this one is sorted out I could make a PR to refactor GetFriendInfoById() out and just use the new APIs will creating a compat layer for classic.  That is you call one API (a thing wrapper the new one) and get new structures and then code deals with that.  In retail the wrapper just calls the existing new API, while in classic the wrapper will output same content as new APIs using old ones and have to build up structs.  Right now the results of GetFriendInfoById() is just a special case bundle of information used in this particular API and I'm not a huge fan of it since it sounds generic.

Cistoran commented:
> > Yo, is it possible to just change line 314 from
> > `- button.name:SetJustifyH("RIGHT")` to
> > `+ button.name:SetJustifyH("LEFT")`?
> > Looks alot better UI wise in my opinion
> > Also, It doesn't return the current zone/area in the UI, so right now it will either show nothing when in another game, app/mobile or it will just show unknow when playing WoW (Picture below).
> > https://imgur.com/eDirjYK
> 
> Made a new commit justifying the dividers to CENTER to match how it works in classic - author may decide to change that to be LEFT instead.
> 
> Think I've fixed issues with zone names/realm names.
> 
> Also updated colors for classic so if you're there you should see class colors properly.

Even though it's justified Center it seems to not be properly centered, could be due to the scroll frame not being aligned properly with the panel frame? 

See image for where center frame should be.

![image](https://user-images.githubusercontent.com/7893023/66265152-afc95f80-e7ce-11e9-8e15-17b42a12b68b.png)

mudohir commented:
I think the alignment isn't perfectly centered because the Game Icon and Invite Button throw it to the left some.  Just eyeballing it if you expand a group it seems to me that the text is centered between the (+/-) button on the left and the game icon.

A more ambitious fix here could port over the previous frames from pre 8.2.5 and have things looking exactly the same, but that's taking on some risk if they change the standard UI - the changes need to be retrofitted into the old stuff to match whatever changes they make and that seems like it would be a sort of nightmare for the addon author potentially.

Mikeprod commented:
I'll review this tonight :) Thanks a lot. @mudohir do you want me to add you as a permanent contributor ?

Mikeprod commented on FriendGroups.lua:
```diff
@@ -62,10 +89,16 @@ local function ClassColourCode(class,table)
 	end
+	local colour = RAID_CLASS_COLORS[class]
+	-- Shaman color is incorrectly shared with pally in the table in classic
+	if WOW_PROJECT_ID == WOW_PROJECT_CLASSIC and class == "SHAMAN" then
```
Thanks for that, I think this might be the root cause of #38

Mikeprod commented on FriendGroups.lua:
```diff
@@ -83,6 +116,120 @@ local function FriendGroups_GetTopButton(offset)
+		end
+		if client == BNET_CLIENT_WOW then
+			local nameColor = FriendGroups_SavedVars.colour_classes and ClassColourCode(class) or FRIENDS_WOW_NAME_COLOR
+			nameText = nameText.." "..nameColor.."("..characterName..coopLabel..characterNameSuffix..")"..FONT_COLOR_CODE_CLOSE
```
Why adding coopLabel ? Looks weird imo

Mikeprod commented:
> I think the alignment isn't perfectly centered because the Game Icon and Invite Button throw it to the left some. Just eyeballing it if you expand a group it seems to me that the text is centered between the (+/-) button on the left and the game icon.
> 
> A more ambitious fix here could port over the previous frames from pre 8.2.5 and have things looking exactly the same, but that's taking on some risk if they change the standard UI - the changes need to be retrofitted into the old stuff to match whatever changes they make and that seems like it would be a sort of nightmare for the addon author potentially.

Yeah centered is the less worse of all the alignments
